test/common/upstream/host_stats_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "envoy/upstream/host_description.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Upstream {
+namespace {
+
+// Verify that counters are sorted by name.
+TEST(HostStatsTest, CountersSortedByName) {
+  HostStats host_stats;
+  auto counters = host_stats.counters();
+  EXPECT_FALSE(counters.empty());
+
+  const absl::string_view* prev_counter_name = nullptr;
+  for (const auto& counter : counters) {
+    if (prev_counter_name) {
```

Comment:
`prev_counter_name != nullptr` per Envoy convention.

You could also use a loop like:
```
for (size_t i = 1; i < counters.size(); ++i) {
  EXPECT_LT(counters[i-1]->name, counters[i]->name);
}
```

Not sure if that's better but it's fewer lines.

Reply:
Yes, that's better.  Thanks for the suggestion.